AutomatedTesting/Gem/PythonTests/editor_test_testing/TestSuite_Main.py

```diff
@@ -0,0 +1,259 @@
+# Copyright (c) Contributors to the Open 3D Engine Project
+# 
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+"""
+This suite contains the tests for editor_test utilities.
+"""
+
+import pytest
+import os
+import sys
+import importlib
+import re
+
+from ly_test_tools import LAUNCHERS
+
+sys.path.append(os.path.dirname(os.path.abspath(__file__)))
+
+from ly_test_tools.o3de.editor_test import EditorSingleTest, EditorSharedTest, EditorTestSuite, Result
+from ly_test_tools.o3de.asset_processor import AssetProcessor
+import ly_test_tools.environment.process_utils as process_utils
+
+import argparse, sys
+
+@pytest.mark.SUITE_main
+@pytest.mark.parametrize("launcher_platform", ['windows_editor'])
+@pytest.mark.parametrize("project", ["AutomatedTesting"])
+class TestEditorTest:
+    
+    args = None 
+    path = None
+    @classmethod
+    def setup_class(cls):
+        TestEditorTest.args = sys.argv.copy()
+        build_dir_arg_index = TestEditorTest.args.index("--build-directory")
+        if build_dir_arg_index < 0:
+            print("Must pass --build-directory argument")
```

Comment:
suggestion: might be worth it to add ```Error``` to this message to aid searching the log.
i also suggest maybe something like this ```Error: Missing expected argument '--build-directory' passed to  TestEditorTest.``` (or whatever makes most sense).